src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -0,0 +1,595 @@
+            TokenTypes.ELIST,
+            TokenTypes.LITERAL_IF,
+            TokenTypes.LITERAL_FOR,
+            TokenTypes.LITERAL_WHILE,
+        };
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public void beginTree(DetailAST root) {
+        variables = new ArrayDeque<>();
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        switch (ast.getType()) {
+            case TokenTypes.DOT:
+                final DetailAST firstDotOrMethodCallAst = getFirstDotOrMethodCallToken(ast);
+                if (firstDotOrMethodCallAst == null
+                        && ast.findFirstToken(TokenTypes.LITERAL_NEW) == null
+                        && ast.findFirstToken(TokenTypes.LITERAL_SUPER) == null
+                        && ast.findFirstToken(TokenTypes.LITERAL_CLASS) == null
+                        && ast.findFirstToken(TokenTypes.LITERAL_THIS) == null) {
+                    checkIdentifierAst(ast.findFirstToken(TokenTypes.IDENT));
+                }
+                break;
+            case TokenTypes.VARIABLE_DEF:
+                if (ast.getParent().getType() != TokenTypes.OBJBLOCK) {
+                    final DetailAST ident = ast.findFirstToken(TokenTypes.IDENT);
+                    final VariableDesc desc = new VariableDesc(ident.getText(), ast);
+                    setScopeOfVariable(ast, desc);
+                    variables.push(desc);
+                }
+                break;
+            case TokenTypes.IDENT:
+                if (!TokenUtil.isOfType(ast.getParent().getType(),
+                        TokenTypes.VARIABLE_DEF, TokenTypes.DOT,
+                        TokenTypes.LITERAL_NEW, TokenTypes.PATTERN_VARIABLE_DEF)) {
+                    checkIdentifierAst(ast);
+                }
+                break;
+            case TokenTypes.LITERAL_INSTANCEOF:
+                final DetailAST patternVariableDef = ast
+                        .findFirstToken(TokenTypes.PATTERN_VARIABLE_DEF);
+                if (patternVariableDef != null) {
+                    final DetailAST identToken = patternVariableDef
+                            .findFirstToken(TokenTypes.IDENT);
+                    final VariableDesc variableDesc =
+                            new VariableDesc(identToken.getText(), patternVariableDef);
+                    setScopeOfPatternVariable(ast, patternVariableDef, variableDesc);
+                    variables.push(variableDesc);
+                }
+                break;
+            default:
+                break;
+        }
+    }
+
+    @Override
+    public void leaveToken(DetailAST ast) {
+        if (TokenUtil.isOfType(ast.getType(),
+                IntStream.concat(Arrays.stream(SCOPES),
+                        Arrays.stream(SCOPES_EXCLUSIVE_TO_PATTERN_VARIABLES)).toArray())) {
```

Comment:
Please create a new field. Combining constant arrays in runtime is weird.

Reply:
removed as I will be writing a separate check for pattern variables